File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int> &nums) {
        vector<vector<int>> ans;
        vector<int> temp(nums.size() + 1);

        for(int i : nums) {
            if(temp[i] >= ans.size())
                ans.push_back({});
            ans[temp[i]].push_back(i);
            temp[i]++;
        }

        return ans;
    }

};

File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        temp = {}
        ans = []

        for num in nums:
            if num not in temp:
                temp[num] = 0

            i = temp[num]

            if i == len(ans):
                ans.append([])

            ans[i].append(num)

            temp[num] += 1

        return ans